tests/e2e/tests/test_clustering.py

```diff
@@ -15,10 +16,11 @@ def get_join_token(
     cluster_node: harness.Instance, joining_node: harness.Instance, *args: str
 ) -> str:
     out = cluster_node.exec(
-        ["k8s", "get-join-token", joining_node.id, *args],
+        ["k8s", "get-join-token", joining_node.id, "--output-format", "json", *args],
         capture_output=True,
     )
-    return out.stdout.decode().strip()
+    result = out.stdout.decode().strip()
+    return json.loads(result)["join-token"]
```

Comment:
the `.strip()` isn't necessary:

```
>>> json.loads('\n\t{"1":1}\n\t')
{'1': 1}
```

```suggestion
    result = json.loads(out.stdout.decode())
    return result["join-token"]
```